core/domain/fs_domain_test.py

```diff
@@ -105,16 +107,108 @@ def test_independence_of_file_systems(self):
             fs2.get('abc.png')
 
 
-class DiskBackedFileSystemTests(test_utils.GenericTestBase):
+class DiskBackedFileSystemUnitTests(test_utils.GenericTestBase):
     """Tests for the disk-backed file system."""
 
+    def setUp(self):
+        super(DiskBackedFileSystemUnitTests, self).setUp()
+        self.user_id = 'abc@example.com'
+
     def test_get(self):
         fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
             feconf.TESTS_DATA_DIR))
         self.assertTrue(fs.get('img.png'))
         with self.assertRaisesRegexp(IOError, 'No such file or directory'):
             fs.get('non_existent_file.png')
 
+    def test_commit(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        with self.assertRaises(NotImplementedError):
+            fs.commit(self.user_id, 'abc.png', 'file_contents')
+
+    def test_isfile(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        self.assertTrue(fs.isfile('img.png'))
+        self.assertFalse(fs.isfile('fake.png'))
+
+    def test_delete(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        with self.assertRaises(NotImplementedError):
+            fs.delete(self.user_id, 'img.png')
+
+    def test_listdir(self):
+        fs = fs_domain.AbstractFileSystem(fs_domain.DiskBackedFileSystem(
+            feconf.TESTS_DATA_DIR))
+        with self.assertRaises(NotImplementedError):
+            fs.listdir('')
+
+
+class GcsFileSystemUnitTests(test_utils.GenericTestBase):
+    """ Tests for the GCS file system."""
+
+    def setUp(self):
+        super(GcsFileSystemUnitTests, self).setUp()
+        self.user_id = 'abc@example.com'
+
+    def test_get_and_save(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            fs = fs_domain.AbstractFileSystem(
```

Comment:
Ditto, regarding moving to setup(). Keep in mind that if we move out, you'll have to swap DEV_MODE while creating.

Reply:
done.